FBShimmering/FBShimmeringLayer.m

```diff
@@ -244,6 +244,22 @@ - (void)setShimmeringOpacity:(CGFloat)shimmeringOpacity
   }
 }
 
+- (CFTimeInterval)shimmeringBeginTime
+{
+  return [_maskLayer animationForKey:kFBShimmerSlideAnimationKey].beginTime;
+}
+
+- (void)resumeShimmeringWithBeginTime:(CFTimeInterval)beginTime
+{
+  CAAnimation *slideAnimation = [[_maskLayer animationForKey:kFBShimmerSlideAnimationKey] mutableCopy];
+  if (nil == slideAnimation) {
+    return;
+  }
+
+  slideAnimation.beginTime = beginTime;
+  [_maskLayer addAnimation:slideAnimation forKey:kFBShimmerSlideAnimationKey];
```

Comment:
Good point, maybe needs a different sentinel value.

Yeah, very true about `CACurrentMediaTime()` — non-linear timescales are rare, but good to support them!